internal/db/arango.go

```diff
@@ -106,9 +107,30 @@ func (c *ArangoClient) QueryAnnotations(ctx context.Context, key string) ([]docu
 	if err != nil {
 		return nil, err
 	}
-	query := "FOR a in annotations FILTER a.dataRef == @key RETURN a"
+	query := `
+			LET stackAnnotations = (
+    				FOR a IN annotations FILTER a.dataRef == @key 
+                                	LET hostAnnotation = (
+						FOR hostAn in annotations 
+						FILTER a.host == hostAn.host 
+							AND (hostAn.layer == @host OR hostAn.layer == @os) 
+				      		RETURN hostAn
+			          	)
+                                        LET tagAnnotation = (
+				        	FOR tagAn IN annotations 
+				      		FILTER a.tag == tagAn.tag AND tagAn.layer == @cicd  
+				      		RETURN tagAn
+			                )
+                                RETURN DISTINCT APPEND(tagAnnotation, hostAnnotation)
+                        )
+			LET appAnnotations = (FOR a IN annotations FILTER a.dataRef == @key RETURN a)
+			RETURN FLATTEN(APPEND(appAnnotations, stackAnnotations))
+			`
```

Comment:
This part of stackAnnotations won't return the host layer annotations because there is no direct relationship between the App layer annotations and Host layer annotations.
Instead, we should use the OS layer annotations to fetch the Host layer annotations.
```suggestion
	query := `
			LET stackAnnotations = (
				FOR a IN annotations 
				FILTER a.dataRef == @key 
				LET osHostAnnotations = (
					FOR osAn IN annotations 
					FILTER a.host == osAn.host AND osAn.layer == @os
					LET hostAnnotations = (
						FOR hostAn IN annotations
						FILTER hostAn.tag == osAn.tag AND hostAn.layer == @host
						RETURN hostAn
					)
					RETURN FLATTEN(APPEND([osAn], hostAnnotations))
				)
				LET cicdAnnotations = (
					FOR tagAn IN annotations 
					FILTER a.tag == tagAn.tag AND tagAn.layer == @cicd  
					RETURN tagAn
				)
				RETURN DISTINCT FLATTEN(APPEND(cicdAnnotations, osHostAnnotations))
			)

			LET appAnnotations = (
				FOR a IN annotations 
				FILTER a.dataRef == @key 
				RETURN a
			)

			RETURN FLATTEN(APPEND(appAnnotations, stackAnnotations))
			`
```